samples/Directory.Build.targets

```diff
@@ -0,0 +1,21 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="_GetBicepTemplates">
+    <ItemGroup>
+      <_BicepTemplate Include="@(None)" Condition="'%(Extension)' == '.bicep'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_BuildBicepTemplates"
+          Condition="'$(BuildBicepTemplates)' != 'false'"
+          DependsOnTargets="_GetBicepTemplates"
+          BeforeTargets="CoreCompile"
+          Inputs="@(_BicepTemplate)"
+          Outputs="@(_BicepTemplate->'%(RelativeDir)%(Filename).json')">
+    <Exec Command="az bicep build --file &quot;@(_BicepTemplate)&quot; --outfile &quot;@(_BicepTemplate->'%(RelativeDir)%(Filename).json')&quot;">
```

Comment:
What's the build experience if the Azure CLI isn't installed?   Does this just fail to generate the ARM template and otherwise not impact the build or would it fail the build?

Reply:
It would fail, but this would only be triggered if the .bicep file is newer than the .json file, which means the contributor made modifications - and hopefully tested them, most likely using the az CLI - or git checked out the .bicep file first and the contributor is building samples specifically.

I feel it's a pretty low likelihood that all of these conditions come together to be a real problem, but we could continue our conversation about a service endpoint instead. I'd like to get this in and need the .json file, but if you'd prefer I could make this behavior opt-in instead.

/cc @pakrym

Reply:
Fair enough;  I don't think that sounds unreasonable.